aspnetcore/performance/memory.md

```diff
@@ -0,0 +1,408 @@
+* With frequent calls, causes app memory to increases until the process crashes with an `OutOfMemory` exception.
+
+![preceding chart](memory/_static/eternal.png)
+
+In the preceding image:
+
+* Load testing the `/api/staticstring` endpoint causes a linear increase in memory.
+* The GC tries to free memory as the memory pressure grows, by calling a generation 2 collection.
+* The GC cannot free the leaked memory. Allocated and working set increase with time.
+
+Some scenarios, such as caching, require object references to be held until memory pressure forces them to be released. The <xref:System.WeakReference> class can be used for this type of caching code. A `WeakReference` object is collected under memory pressures. The default implementation of <xref:Microsoft.Extensions.Caching.Memory.IMemoryCache> uses `WeakReference`.
+
+#### Native memory
+
+Some .NET Core objects rely on native memory. Native memory can **not** be collected by the GC. The .NET object using native memory must free it using native code.
+
+.NET provides the <xref:System.IDisposable> interface to let developers release native memory. Even if <xref:System.IDisposable.Dispose*> is not called, correctly implemented classes call `Dispose` when the [finalizer](/dotnet/csharp/programming-guide/classes-and-structs/destructors) runs.
+
+Consider the following code:
+
+```csharp
+[HttpGet("fileprovider")]
+public void GetFileProvider()
+{
+    var fp = new PhysicalFileProvider(TempPath);
+    fp.Watch("*.*");
+}
+```
+
+[PhysicaFileProvider](/dotnet/api/microsoft.extensions.fileproviders.physicalfileprovider?view=dotnet-plat-ext-3.0) is a managed class, so any instance will be collected at the end of the request.
+
+Here is the resulting memory profile while invoking this API continuously.
+
+![preceding chart](memory/_static/fileprovider.png)
+
+The preceding chart shows an obvious issue with the implementation of this class, as it keeps increasing memory usage. This is a known problem that is being tracked in [this issue](https://github.com/aspnet/Home/issues/3110).
+
+The same leak could be happen in user code, by one of the following:
+
+* Not releasing the class correctly.
+* Forgetting to invoke the `Dispose`method of the dependent objects that should be disposed.
+
+#### Large objects heap
+
+Frequent memory allocation/free cycles can fragment memory, especially when allocating large chunks of memory. Objects are allocated in contiguous blocks of memory. To mitigate fragmentation, when the GC frees memory, it will try to defragment it. This process is called **compaction**. Compaction involves moving objects. Moving large objects imposes a performance penalty. For this reason the GC creates a special memory zone for  _large_ objects, called the [large object heap](/dotnet/standard/garbage-collection/large-object-heap) (LOH). Object that are greater than 85,000 bytes (approximately 83 KB) are:
+
+* Placed on the LOH.
+* Not compacted.
```

Comment:
>Not compacted. [](start = 2, length = 14)

LOH can be compacted at the request of the user;
in containers LOH is automatically compacted starting core 3.0